module/regenerator.py

```diff
@@ -138,16 +147,29 @@ def load_from_scheduler(self, sched):
         self.contactgroups = c.contactgroups
         self.timeperiods = c.timeperiods
         self.commands = c.commands
+        # WebUI - Manage notification ways
         self.notificationways = c.notificationways
+
         # We also load the realm
         for h in self.hosts:
+            # WebUI - Manage realms if declared (Alignak)
             if getattr(h, 'realm_name', None):
                 self.realms.add(h.realm_name)
+            else:
+                # WebUI - Manage realms if declared (Shinken)
+                if getattr(h, 'realm', None):
+                    self.realms.add(h.realm)
+            # WebUI - be aware that the realm may be a string or an object
+            # This will be managed later.
             break
```

Comment:
```python
if self.hosts:
    self.realms.add(getattr(self.hosts[0], 'realm_name',
                                         default=getattr(self.hosts[0], 'realm', default=None))
```

Reply:
Hmmm... not really correct. We do not want only the realm of the first host. But I agree with your more compact code writing form.
I used anf if / else to point out the difference between Shinken and Alignak 😉

Reply:
With this break you already only get the realm of the first host.

This is the code in Shinken's regenerator :

```

Reply:
Yes, but it is buggy! There is not interest to get only the first host realm ... what if you have several realms in your configuration. Anyway, this part of the source code is never used (and has never been used!) because it only concerns laoding the WebUI module in a scheduler!